### PR TITLE
Eejt 0001

### PR DESCRIPTION
**Pull request by Oleh-Shynkarchuk**
Task is completed.

### CODE REVIEW

File: multi_project/buildSrc/src/main/groovy/multi_project.java-common-conventions.gradle [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+repositories {
+    // Use Maven Central for resolving dependencies.
+    mavenCentral()
+}
+
+dependencies {
+    constraints {
+        // Define dependency versions as constraints
+        implementation 'org.apache.commons:commons-text:1.9'
```
Comment (OlehHoliuk) [inline multi_project/buildSrc/src/main/groovy/multi_project.java-common-conventions.gradle:20]: Why do we need this?

File: multi_project/api/src/main/java/multi_project/api/App.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ */
+package multi_project.api;
+
+
+import multi_project.core.Utils;
+
+public class App {
+    public static void main(String[] args) {
+        System.out.println(Utils.isAllPositiveNumbers("12", "79"));
+        Utils.isAllPositiveNumbers(args);
```
Comment (OlehHoliuk) [inline multi_project/api/src/main/java/multi_project/api/App.java:12]: Why do we need the second call with args?

File: multi_project/core/build.gradle [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    id 'multi_project.java-application-conventions'
+}
+
+dependencies {
+    implementation files('customlib/utils-1.3.5.jar')
+}
+
+application {
+    // Define the main class for the application.
+    mainClass = 'multi_project.api.App'
```
Comment (OlehHoliuk) [inline multi_project/core/build.gradle:17]: Why do we need this definition inside `core` module?

File: multi_project/core/src/main/java/multi_project/core/Utils.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+public class Utils {
+    public static boolean isAllPositiveNumbers(String... str) {
+        boolean isPositive = true;
+        for (String s : str) {
+            if (!StringUtils.isPositiveNumber(s)) {
+                isPositive = false;
+            }
+        }
+        return isPositive;
```
Comment (OlehHoliuk) [inline multi_project/core/src/main/java/multi_project/core/Utils.java:16]: Rewrite this using Stream API. Use `allMatch` method

**Review by OlehHoliuk — CHANGES_REQUESTED** [review]

File: multi_project/buildSrc/build.gradle [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+plugins {
+    // Support convention plugins written in Groovy. Convention plugins are build scripts in 'src/main' that automatically become available as plugins in the main build.
+    id 'groovy-gradle-plugin'
+}
+
+repositories {
+    // Use the plugin portal to apply community plugins in convention plugins.
+    gradlePluginPortal()
+}
```
Comment (OlehHoliuk) [inline multi_project/buildSrc/build.gradle:15]: What do we need `buildSrc` for?

**Review by OlehHoliuk — CHANGES_REQUESTED** [review]

File: multi_project/buildSrc/src/main/groovy/multi_project.java-common-conventions.gradle (hunk 1 above)
Comment (Oleh-Shynkarchuk) [reply multi_project/buildSrc/src/main/groovy/multi_project.java-common-conventions.gradle]: gradle init add this library. Removed

File: multi_project/api/src/main/java/multi_project/api/App.java (hunk 2 above)
Comment (Oleh-Shynkarchuk) [reply multi_project/api/src/main/java/multi_project/api/App.java]: repeat(to self) how main(args) works. removed

File: multi_project/core/build.gradle (hunk 3 above)
Comment (Oleh-Shynkarchuk) [reply multi_project/core/build.gradle]: idk, gradle init created it, project works wout it, so i remove him

File: multi_project/core/src/main/java/multi_project/core/Utils.java (hunk 4 above)
Comment (Oleh-Shynkarchuk) [reply multi_project/core/src/main/java/multi_project/core/Utils.java]: done

File: multi_project/buildSrc/build.gradle (hunk 5 above)
Comment (Oleh-Shynkarchuk) [reply multi_project/buildSrc/build.gradle]: gradle init multiproject aplication. Rewrite wout this module

**Pull request merged**